app/views/service_worker/index.js.erb

```diff
@@ -37,40 +37,45 @@
   function createPageStream(request) {
     const stream = new ReadableStream({
       start(controller) {
-        if (!caches.match('/shell_top') || !caches.match('/shell_bottom')) { //return if shell isn't cached.
-          return
-        }
+        Promise.all([caches.match('/shell_top'), caches.match('/shell_bottom')])
+          .then((cachedShellMatches) => {
+            const cachedShellTop = cachedShellMatches[0];
+            const cachedShellBottom = cachedShellMatches[1];
+            if (!cachedShellTop || !cachedShellBottom) { // return if shell isn't cached.
+              return
```

Comment:
Note that this codepath has actually never been reached before. I don't know if it could cause mega explosions... all I know is that I tested loading `dev.to` locally with my changes, and it still loaded. But that's all I know, hear?

Reply:
This is true that this code path is never hit and that all the `catches.match(...)` calls are promises. However, it goes deeper than that as there are other checks in the service worker code that are also Promises that always return true because they are not being handled as Promises. See https://github.com/thepracticaldev/dev.to/blob/master/app/views/service_worker/index.js.erb#L131

I think this fix is a little more involved and we definitely need to be able to test this. cc: @benhalpern

Reply:
so i hit the bug again today. 

interestingly, i tried reproducing it, this time by executing `caches.delete` for both `shell_top` and `shell_bottom`, in the main console as well as the js console... it did not reproduce the bug! so strange.